### PR TITLE
Added bulk boosting prediction functions

### PR DESCRIPTION
**Pull request by yassineSaid-craft**

### CODE REVIEW

File: craft_ai/client.py [hunk 1]
```diff
@@ -1203,6 +1203,48 @@ def _get_entity_boosting_decision(self, entity_id, window, entity_type, context)
...
 
+    def get_agent_bulk_boosting_decision(self, payload):
+        """Get a group of boosting decisions.
+
+        :param list payload: Contains the informations to get the decisions.
+        It's in the form [{"entityName": agent_id, "timeWindow": [from, to], "context": context}]
+        With timeWindow and context having the form given in the craft_ai documentation.
+
+        :return: list of decisions.
+        :rtype: List of dict.
```
Comment (valerianknk) [inline craft_ai/client.py:1214]:
```suggestion
        :rtype: list of dict.
```

File: craft_ai/client.py [hunk 2]
```diff
@@ -1203,6 +1203,48 @@ def _get_entity_boosting_decision(self, entity_id, window, entity_type, context)
 
         return decision
 
+    def get_agent_bulk_boosting_decision(self, payload):
+        """Get a group of boosting decisions.
+
+        :param list payload: Contains the informations to get the decisions.
+        It's in the form [{"entityName": agent_id, "timeWindow": [from, to], "context": context}]
+        With timeWindow and context having the form given in the craft_ai documentation.
```
Comment (valerianknk) [inline craft_ai/client.py:1211]: Maybe add the URL pointing to the specified documentation

File: tests/test_get_bulk_agent_boosting_decision.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        )
+        self.client.delete_agent(self.agent_id_2)
+        self.client.create_agent(
+            valid_data.VALID_BOOSTING_CONFIGURATION, self.agent_id_2
+        )
+        self.client.add_agent_operations(
+            self.agent_id_2, valid_data.VALID_OPERATIONS_SET_COMPLETE_1
+        )
+
+    def clean_up_agent(self, aid):
```
Comment (valerianknk) [inline tests/test_get_bulk_agent_boosting_decision.py:36]: You should use the _tearDown_ method for fixtures deletion. The method will be called method after each test.

File: tests/test_get_bulk_agent_boosting_decision.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
```
Comment (valerianknk) [inline tests/test_get_bulk_agent_boosting_decision.py:98]: Same as previously, you can rename _clean_up_agent_ into tearDown and remove clean_up_agent calls.

File: tests/test_get_bulk_agent_boosting_decision.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        # Makes sure that no agent with the standard ID remains
+        self.client.delete_agent(aid)
+
+    def test_get_bulk_decision_boosting_with_non_existant_agent_id(self):
+        """get_agent_bulk_boosting_decision should handle valid and non existant
+        agents"""
+        first_timestamp = valid_data.VALID_OPERATIONS_SET_COMPLETE_1[0]["timestamp"]
+        last_timestamp = valid_data.VALID_OPERATIONS_SET_COMPLETE_1[-1]["timestamp"]
+        timeWindow = [first_timestamp, last_timestamp]
```
Comment (valerianknk) [inline tests/test_get_bulk_agent_boosting_decision.py:107]:
```suggestion
        timeWindow = [first_timestamp, last_timestamp]
        
```

File: tests/test_get_bulk_agent_boosting_decision.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+                "timeWindow": timeWindow,
+                "context": context,
+            },
+            {
+                "entityName": self.agent_id_2,
+                "timeWindow": timeWindow,
+                "context": context,
+            },
+            {"entityName": "test_test", "timeWindow": timeWindow, "context": context},
+        ]
```
Comment (valerianknk) [inline tests/test_get_bulk_agent_boosting_decision.py:121]:
```suggestion
        ]
        
```

File: tests/test_get_bulk_generator_boosting_decision.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+            self.agent_id, valid_data.VALID_OPERATIONS_SET_COMPLETE_1
+        )
+        BOOSTING_CONFIGURATION = valid_data.VALID_GENERATOR_BOOSTING_CONFIGURATION
+        BOOSTING_CONFIGURATION["filter"] = [self.agent_id]
+        self.client.delete_generator(self.generator_id_1)
+        self.client.delete_generator(self.generator_id_2)
+        self.client.create_generator(BOOSTING_CONFIGURATION, self.generator_id_1)
+        self.client.create_generator(BOOSTING_CONFIGURATION, self.generator_id_2)
+
+    def clean_up_agent(self, aid):
```
Comment (valerianknk) [inline tests/test_get_bulk_generator_boosting_decision.py:38]:
```suggestion
    def tearDown(self:
```

File: tests/test_get_bulk_generator_boosting_decision.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        # Makes sure that no agent with the standard ID remains
+        self.client.delete_agent(aid)
+
+    def test_get_bulk_decision_boosting_with_correct_input(self):
+        first_timestamp = valid_data.VALID_OPERATIONS_SET_COMPLETE_1[0]["timestamp"]
+        last_timestamp = valid_data.VALID_OPERATIONS_SET_COMPLETE_1[-1]["timestamp"]
+        timeWindow = [first_timestamp, last_timestamp]
```
Comment (valerianknk) [inline tests/test_get_bulk_generator_boosting_decision.py:45]:
```suggestion
        timeWindow = [first_timestamp, last_timestamp]
        
```

File: tests/test_get_bulk_generator_boosting_decision.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+                "entityName": self.generator_id_1,
+                "timeWindow": timeWindow,
+                "context": context,
+            },
+            {
+                "entityName": self.generator_id_2,
+                "timeWindow": timeWindow,
+                "context": context,
+            },
+        ]
```
Comment (valerianknk) [inline tests/test_get_bulk_generator_boosting_decision.py:58]:
```suggestion
        ]
        
```

File: tests/test_get_bulk_generator_boosting_decision.py [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    def clean_up_agent(self, aid):
+        # Makes sure that no agent with the standard ID remains
+        self.client.delete_agent(aid)
+
+    def test_get_bulk_decision_boosting_with_non_existant_generator_id(self):
+        """get_agent_bulk_boosting_decision should handle valid and non existant
+        agents"""
+        first_timestamp = valid_data.VALID_OPERATIONS_SET_COMPLETE_1[0]["timestamp"]
+        last_timestamp = valid_data.VALID_OPERATIONS_SET_COMPLETE_1[-1]["timestamp"]
+        timeWindow = [first_timestamp, last_timestamp]
```
Comment (valerianknk) [inline tests/test_get_bulk_generator_boosting_decision.py:112]:
```suggestion
        timeWindow = [first_timestamp, last_timestamp]
        
```

File: tests/test_get_bulk_generator_boosting_decision.py [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+                "timeWindow": timeWindow,
+                "context": context,
+            },
+            {
+                "entityName": self.generator_id_2,
+                "timeWindow": timeWindow,
+                "context": context,
+            },
+            {"entityName": "test_test", "timeWindow": timeWindow, "context": context},
+        ]
```
Comment (valerianknk) [inline tests/test_get_bulk_generator_boosting_decision.py:126]:
```suggestion
        ]
        
```

File: craft_ai/client.py [hunk 12]
```diff
@@ -1203,6 +1203,48 @@ def _get_entity_boosting_decision(self, entity_id, window, entity_type, context)
...
+
+    def get_generator_bulk_boosting_decision(self, payload):
+        """Get a group of boosting decisions.
+
+        :param list payload: Contains the informations to get the decisions. It's in the form
+        [{"entityName": generator_id, "timeWindow": [from, to], "context": context}]
+        With timeWindow and context having the form given in the craft_ai documentation.
+
+        :return: list of decisions.
+        :rtype: List of dict.
```
Comment (valerianknk) [inline craft_ai/client.py:1234]:
```suggestion
        :rtype: list of dict.
```

File: tests/test_get_bulk_agent_boosting_decision.py [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+    two valid agents"""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.client = Client(settings.CRAFT_CFG)
+        cls.agent_id_1 = generate_entity_id("test_get_agent_bulk_boosting_decision")
+        cls.agent_id_2 = generate_entity_id("test_get_agent_bulk_boosting_decision")
+
+    def setUp(self):
+        self.client.delete_agent(self.agent_id_1)
```
Comment (valerianknk) [inline tests/test_get_bulk_agent_boosting_decision.py:21]: The deletion will take place in the _tearDown_ method so you can remove the deletion here

File: tests/test_get_bulk_agent_boosting_decision.py [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+    with invalid parameters for agents"""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.client = Client(settings.CRAFT_CFG)
+        cls.agent_id_1 = generate_entity_id("test_get_agent_bulk_boosting_decision")
+        cls.agent_id_2 = generate_entity_id("test_get_agent_bulk_boosting_decision")
+
+    def setUp(self):
+        self.client.delete_agent(self.agent_id_1)
```
Comment (valerianknk) [inline tests/test_get_bulk_agent_boosting_decision.py:83]: Same as previously

File: tests/test_get_bulk_generator_boosting_decision.py [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+        cls.agent_id = generate_entity_id("test_get_generator_bulk_boosting_decision")
+        cls.generator_id_1 = generate_entity_id(
+            "test_get_generator_bulk_boosting_decision"
+        )
+        cls.generator_id_2 = generate_entity_id(
+            "test_get_generator_bulk_boosting_decision"
+        )
+
+    def setUp(self):
+        self.client.delete_agent(self.agent_id)
```
Comment (valerianknk) [inline tests/test_get_bulk_generator_boosting_decision.py:26]: Same, deletion in _tearDown_ method

File: craft_ai/client.py (hunk 2 above)
Comment (yassineSaid-craft) [reply craft_ai/client.py:1211]: There are no links to the documentation in the other functions

**Review by valerianknk — APPROVED** [review]

**Pull request merged**